File: Grammar/modules/Parsers/JavaCyk/src/main/java/CellRule.java
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

@Data
public class CellRule {

    @JsonProperty
    private Rule rule;

    @JsonProperty
    private Coordinate cell1Coordinates;

    @JsonProperty
    private Coordinate cell2Coordinates;

    @JsonProperty
    private Coordinate rootCellCoordinates;

    public CellRule() {

    }

    public CellRule(Rule rule) {
        this.rule = rule;
    }

    public CellRule(Rule rule, Coordinate c1, Coordinate c2) {
        this.rule = rule;
        this.cell1Coordinates = c1;
        this.cell2Coordinates = c2;
    }

}

File: Grammar/modules/GCSBase/domain/CellRule.py
from ..domain import Coordinates
from ..domain.Rule import Rule


class CellRule:
    def __init__(self, rule: Rule = None, c1: Coordinates = None, c2: Coordinates = None):
        self.rule = rule
        self.cell_2_coordinates: Coordinates = c2
        self.cell_1_coordinates: Coordinates = c1
        self.root_cell_coordinates = None
        self.used_in_parsing = False
        self.called_number = 0
        self.tmp_val = 0.0
        self.parsed = False
        self.index = None
        self.proceeded = False

    def __eq__(self, other):
        if type(self) is type(other):
            return self.rule == other.rule
        return False

    def __hash__(self) -> int:
        return hash(self.rule)
